scripts/node_merge.sh

```diff
@@ -0,0 +1,218 @@
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+DST_ROOT_FOLDER="$1"
+SRC_ROOT_FOLDER="$2"
+
+DST_CLUSTER_LOCK_FILE="$DST_ROOT_FOLDER/$CLUSTER_LOCK_FILENAME"
+SRC_CLUSTER_LOCK_FILE="$SRC_ROOT_FOLDER/$CLUSTER_LOCK_FILENAME"
+
+DST_KEYS_FOLDER="$DST_ROOT_FOLDER/$VALIDATOR_KEYS_DIRNAME"
+SRC_KEYS_FOLDER="$SRC_ROOT_FOLDER/$VALIDATOR_KEYS_DIRNAME"
+
+# --- Part 1: Merge cluster-lock.json ---
+
+# Check if required JSON files exist
+if [ ! -f "$DST_CLUSTER_LOCK_FILE" ]; then
+    echo "Error: Destination file '$DST_CLUSTER_LOCK_FILE' not found."
+    exit 1
+fi
+if [ ! -f "$SRC_CLUSTER_LOCK_FILE" ]; then
+    echo "Error: Source file '$SRC_CLUSTER_LOCK_FILE' not found."
+    exit 1
+fi
+
+# Check file permissions for cluster-lock.json
+if [ ! -r "$SRC_CLUSTER_LOCK_FILE" ]; then
+    echo "Error: Source file '$SRC_CLUSTER_LOCK_FILE' is not readable. Please check permissions."
+    exit 1
+fi
+if [ ! -r "$DST_CLUSTER_LOCK_FILE" ]; then
+    echo "Error: Destination file '$DST_CLUSTER_LOCK_FILE' is not readable. Please check permissions."
+    exit 1
+fi
+if [ ! -w "$DST_CLUSTER_LOCK_FILE" ]; then
+    echo "Error: Destination file '$DST_CLUSTER_LOCK_FILE' is not writable. Please check permissions."
+    exit 1
+fi
+
+# Check if jq is installed
+if ! command -v jq &>/dev/null; then
+    echo "Error: 'jq' is not installed. Please install it to use this script."
+    echo "  On Debian/Ubuntu: sudo apt-get install jq"
+    echo "  On macOS (Homebrew): brew install jq"
+    exit 1
+fi
+
+echo "Merging '$CLUSTER_LOCK_FILENAME' from '$SRC_CLUSTER_LOCK_FILE' into '$DST_CLUSTER_LOCK_FILE'..."
+
+# Use jq to perform the merge
+# We will stream the destination JSON into jq's stdin, and load the source JSON
+# as a separate argument using --slurpfile.
+# Since --slurpfile reads the file into an array, we'll access the first element ([0])
+# of the $src_data variable.
+MERGED_JSON=$(
+    jq \
+        --slurpfile src_data "$SRC_CLUSTER_LOCK_FILE" \
+        '
+    .cluster_definition.num_validators = (.cluster_definition.num_validators + ($src_data[0].cluster_definition.num_validators)) |
+    .cluster_definition.validators = (.cluster_definition.validators + $src_data[0].cluster_definition.validators) |
+    .distributed_validators = (.distributed_validators + $src_data[0].distributed_validators)
+    ' "$DST_CLUSTER_LOCK_FILE"
+)
+
+# Write the merged JSON back to the destination file
+# It's safer to write to a temporary file first and then move it,
+# to prevent data loss if the write operation is interrupted.
+TEMP_DST_JSON_FILE=$(mktemp "${DST_CLUSTER_LOCK_FILE}.tmp.XXXXXX")
+echo "$MERGED_JSON" >"$TEMP_DST_JSON_FILE"
+
+# Check if the temporary file was written successfully
+if [ $? -ne 0 ]; then
+    echo "Error: Failed to write to temporary file '$TEMP_DST_JSON_FILE'."
+    rm -f "$TEMP_DST_JSON_FILE" # Clean up temp file
+    exit 1
+fi
+
+# Move the temporary file to overwrite the original destination file
+mv "$TEMP_DST_JSON_FILE" "$DST_CLUSTER_LOCK_FILE"
```

Comment:
Isn't it better to do this after we successful part 2 as well?
Or we can create a trap function to bring back the original if something in part 2 fails?